### PR TITLE
Feature/thenmozhi - Binded eventtimelinewidget with eventdetails page

### PR DESCRIPTION
**Pull request by tsubramaniam**
completed event details page with eventtimeline widget binding.
Review the code and let me know if it need any changes.

### CODE REVIEW

File: packages/ui/src/components/eventDetails.js [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+class EventDetails extends Component
+{
+  componentDidMount =  () => {
+     this.props.getEventDetails(this.props.match.params.id,this.props.userInfo.id);
+  }
+  constructor(props){
+    super(props);
+  }
+
+  renderBreadCrumb = () => {
```
Comment (shalinijha92) [inline packages/ui/src/components/eventDetails.js]: Please add proper comments for every function

Comment (shalinijha92) [inline packages/ui/src/components/eventDetails.js]: Please add JSDocs(comments) for every function

File: packages/ui/src/components/eventDetails.js [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+    console.log(this.props);
+    return this.state.BreadCrumb.map((item, index) => {
+        return (<li key={index}> <a href={item.Url}> {item.Title} </a> </li>);
+    })
+  }
+
+  render()
+  {
+    console.log(JSON.stringify(this.props)+"props");
```
Comment (shalinijha92) [inline packages/ui/src/components/eventDetails.js]: Remove consoles from UI end when you push it

File: packages/ui/src/components/eventDetails.js [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+  componentDidMount =  () => {
+     this.props.getEventDetails(this.props.match.params.id,this.props.userInfo.id);
+  }
+  constructor(props){
+    super(props);
+  }
+
+  renderBreadCrumb = () => {
+    console.log(this.props);
+    return this.state.BreadCrumb.map((item, index) => {
```
Comment (shalinijha92) [inline packages/ui/src/components/eventDetails.js]: Does this even work? Where is the state variable? you are supposed to declare and initialize them in constructor

File: packages/ui/src/components/eventTimelineWidget.js [hunk 4]
```diff
@@ -30,30 +31,41 @@ class EventTimelineWidget extends Component {
     }
 
     /**
+     * @name loadEventDetails
+     * @desc Move to event details page
+     * @return Loads the event details page
+     */
+    loadEventDetails = (id) => {
+      console.log(id)
+      History.push(`/eventDetails/${id}`)
```
Comment (shalinijha92) [inline packages/ui/src/components/eventTimelineWidget.js]: This is wrong. You are not supposed to route it this way. these routes should be done by Link

File: packages/ui/src/components/eventTimelineWidget.js [hunk 5]
```diff
@@ -30,30 +31,41 @@ class EventTimelineWidget extends Component {
...
      * @desc Iterates through the list of the events and renders the list of events
      * @return Rendered events list {HTML}
      */
     renderEvents = () => {
         return this.props.events.map((event, index) => {
-            return <ListItem 
-            key={index} 
+            return <ListItem
+            key={index}
             primaryText={event.title}
```
Comment (shalinijha92) [inline packages/ui/src/components/eventTimelineWidget.js]: Replace primary text to <Link to={routepath}>{event.title}</Link>

Refer https://reacttraining.com/react-router/web/example/auth-workflow

File: packages/ui/src/containers/eventDetailsContainer/reducer.js [hunk 6]
```diff
@@ -0,0 +1,18 @@
...
+ * @author Thenmozhi Subramaniam
+ * @description A reducer function that returns the event details
+ * based on the action type passed to it
+ * @param {Object} state
+ * @param {Object} action
+ */
+
+import { GET_EVENTDETAILS } from "../../constants/actions";
+export default function (state = [], action) {
+    console.log('In', action.type)
```
Comment (shalinijha92) [inline packages/ui/src/containers/eventDetailsContainer/reducer.js]: Should we store event detail in reducer? I don't think this is required

File: packages/ui/src/css/app.css [hunk 7]
```diff
@@ -37,3 +37,22 @@
   from { transform: rotate(0deg); }
   to { transform: rotate(360deg); }
 }
+
+.Event-details{
+  padding:20px;
+}
+.Event-details > .Date
```
Comment (shalinijha92) [inline packages/ui/src/css/app.css]: classname should start with small letter

File: packages/ui/src/components/eventTimelineWidget.js (hunk 4 above)
Comment (tsubramaniam) [reply packages/ui/src/components/eventTimelineWidget.js]: Made changes as per you comment

File: packages/ui/src/containers/eventDetailsContainer/reducer.js (hunk 6 above)
Comment (tsubramaniam) [reply packages/ui/src/containers/eventDetailsContainer/reducer.js]: Yes, its not required to be stored in reducer, changed the logic to store the values in state.

**Pull request merged**